src/StreamSelectLoop.php

```diff
@@ -220,27 +213,117 @@ private function waitForStreamActivity($timeout)
     }
 
     /**
+     * Returns integer amount of seconds in $time.
+     *
+     * @param float $time – time in seconds
+     *
+     * @return int
+     */
+    private static function getSeconds($time)
+    {
+        /*
+         * intval(floor($time)) will produce int overflow
+         * if $time is (float)PHP_INT_MAX:
+         * (float)PHP_INT_MAX      == PHP_INT_MAX  //true
+         * (int)(float)PHP_INT_MAX == PHP_INT_MAX  //false
+         * (int)(float)PHP_INT_MAX == PHP_INT_MIN  //true
+         * -----------------------------------------------
+         * Loose comparision is intentional, cause $time
+         * is float and
+         * (float)PHP_INT_MAX !== PHP_INT_MAX
+         */
+        if ($time == PHP_INT_MAX) {
+            return PHP_INT_MAX;
+        }
+
+        return intval(floor($time));
+    }
+
+    /**
+     * Returns integer amount of microseconds in $time.
+     *
+     * @param float $time – time in seconds
+     *
+     * @return int
+     */
+    private static function getMicroseconds($time)
+    {
+        $fractional = fmod($time, 1);
+        $microseconds = round($fractional * self::MICROSECONDS_PER_SECOND);
+
+        return intval($microseconds);
+    }
+
+    /**
+     * Returns integer amount of nanoseconds in $time.
+     * The precision is 1 microsecond.
+     *
+     * @param float $time – time in seconds
+     *
+     * @return int
+     */
+    private static function getNanoseconds($time)
+    {
+        return intval(self::getMicroseconds($time) * self::NANOSECONDS_PER_MICROSECOND);
+    }
+
+    /**
      * Emulate a stream_select() implementation that does not break when passed
      * empty stream arrays.
      *
      * @param array        &$read   An array of read streams to select upon.
      * @param array        &$write  An array of write streams to select upon.
-     * @param integer|null $timeout Activity timeout in microseconds, or null to wait forever.
+     * @param float|null   $timeout Activity timeout in seconds, or null to wait forever.
```

Comment:
This is a rather subtle BC break as consumers *could* possibly rely on this `protected` API. Not saying this is an issue with your proposed change, it's more that we ~screwed up~ could have done better with defining our external API.